### PR TITLE
finished homework:

### PR DESCRIPTION
**Pull request by tarkaiev**
added
-registration
-all users
-deleting users
-all products
-adding product to cart
-all products in cart

### CODE REVIEW

**Review by andriylagun — APPROVED** [review]
Good JOB!

File: src/main/java/ecommerce/controllers/AddToCartController.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import ecommerce.model.ShoppingCart;
+import ecommerce.service.interfaces.ProductService;
+import ecommerce.service.interfaces.ShoppingCartService;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/buyProduct")
```
Comment (Sofasmile) [inline src/main/java/ecommerce/controllers/AddToCartController.java]: https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_16_1.html

File: src/main/java/ecommerce/controllers/AddToCartController.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+public class AddToCartController extends HttpServlet {
```
Comment (Sofasmile) [inline src/main/java/ecommerce/controllers/AddToCartController.java]:
```suggestion
public class AddProductToCartController extends HttpServlet {
```

File: src/main/java/ecommerce/controllers/FillerController.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import ecommerce.model.Product;
+import ecommerce.service.interfaces.ProductService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/filler")
```
Comment (Sofasmile) [inline src/main/java/ecommerce/controllers/FillerController.java]:
```suggestion
@WebServlet("/inject")
```

File: src/main/java/ecommerce/controllers/FillerController.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+public class FillerController extends HttpServlet {
```
Comment (Sofasmile) [inline src/main/java/ecommerce/controllers/FillerController.java]:
```suggestion
public class InjectDataController extends HttpServlet {
```

File: src/main/java/ecommerce/controllers/FillerController.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    private static final Injector injector = Injector.getInstance("ecommerce");
+    private final ProductService productService
+            = (ProductService) injector.getInstance(ProductService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+
```
Comment (Sofasmile) [inline src/main/java/ecommerce/controllers/FillerController.java]: let's remove empty line

File: src/main/java/ecommerce/controllers/ProductsController.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/product/all")
+public class ProductsController extends HttpServlet {
```
Comment (Sofasmile) [inline src/main/java/ecommerce/controllers/ProductsController.java]:
```suggestion
public class GetAllProductsController extends HttpServlet {
```

File: src/main/java/ecommerce/controllers/UsersController.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/user/all")
+public class UsersController extends HttpServlet {
```
Comment (Sofasmile) [inline src/main/java/ecommerce/controllers/UsersController.java]:
```suggestion
public class GetAllUsersController extends HttpServlet {
```

**Review by Sofasmile — COMMENTED** [review]
how about: "добавление товара"?

File: src/main/java/ecommerce/controllers/DeleteUserController.java [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+@WebServlet("/deleteUser")
+public class DeleteUserController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("ecommerce");
+    private final UserService userService = (UserService) injector.getInstance(UserService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws IOException {
+        Long userId = Long.valueOf(req.getParameter("userId"));
+        userService.delete(userId);
```
Comment (Sofasmile) [inline src/main/java/ecommerce/controllers/DeleteUserController.java:23]: let's remove the shopping cart too

File: src/main/java/ecommerce/controllers/AddToCartController.java (hunk 1 above)
Comment (tarkaiev) [reply src/main/java/ecommerce/controllers/AddToCartController.java]: fixed

File: src/main/java/ecommerce/controllers/FillerController.java (hunk 3 above)
Comment (tarkaiev) [reply src/main/java/ecommerce/controllers/FillerController.java]: done

File: src/main/java/ecommerce/controllers/FillerController.java (hunk 4 above)
Comment (tarkaiev) [reply src/main/java/ecommerce/controllers/FillerController.java]: done

File: src/main/java/ecommerce/controllers/FillerController.java (hunk 5 above)
Comment (tarkaiev) [reply src/main/java/ecommerce/controllers/FillerController.java]: ok

File: src/main/java/ecommerce/controllers/ProductsController.java (hunk 6 above)
Comment (tarkaiev) [reply src/main/java/ecommerce/controllers/ProductsController.java]: done

File: src/main/java/ecommerce/controllers/DeleteUserController.java (hunk 8 above)
Comment (tarkaiev) [reply src/main/java/ecommerce/controllers/DeleteUserController.java:23]: done

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**